torch/csrc/jit/script/compiler.cpp

```diff
@@ -565,8 +565,14 @@ static std::shared_ptr<SugaredValue> tryEmitBuiltin(
     num_outputs = *value;
   }
 
-  for(size_t i = 0; i < num_outputs; ++i)
-    n->addOutput();
+  for(size_t i = 0; i < num_outputs; ++i) {
+    if (n->kind() == aten::chunk) {
+      // special case for chunk as the schema return only have one Tensor[] type
+      n->addOutput()->setType(DynamicType::get());
```

Comment:
What's the issue with schema having one `Tensor[]` type? Why can't we use it here, and use `Dynamic`, which is incorrect?

Also, another edge case like this is `split`.

Reply:
I think right now we are not support returning list of tensors in the script, `split` is also a case we could not support as for now, it needs to wait until we have mutable list implemented (issue tracked here: https://github.com/pytorch/pytorch/issues/8528). Right now for chunk we returns multiple outputs where each outputs is a DynamicType. I will mark a TODO here to fix the special cases after we have list implemented.

Reply:
Ohh ok. I now see that we infer `num_outputs` elsewhere and inline the list in the IR.